### PR TITLE
ctest: move all tools to ctest regression

### PR DESCRIPTION
**Pull request by gadfort**
Changes:
- removes the use of the previous `regression_tests.tcl`
- Switches ctests generation to build both tcl and python independently.

Notes:
- flow tests are not converted, but should not require much additional effort to migrate (this would be needed to retire the entire tcl based system).
- rmp needed some modifications since both tcl and python scripts edit the same files.

### CODE REVIEW

**Comment by maliberty** [comment]
@gadfort conflicts to resolve.

**Comment by gadfort** [comment]
@maliberty conflict resolved

File: Jenkinsfile
Comment (maliberty) [inline Jenkinsfile:1]: @vvbandeira any concerns here?

**Review by maliberty — APPROVED** [review]

File: Jenkinsfile
Comment (vvbandeira) [reply Jenkinsfile:1]: LGTM

**Pull request merged**